lib/trino-parquet/src/main/java/io/trino/parquet/ParquetMetadataConverter.java

```diff
@@ -155,6 +156,7 @@ public static LogicalTypeAnnotation getLogicalTypeAnnotation(LogicalType type)
                 yield timestampType(timestamp.isAdjustedToUTC, convertTimeUnit(timestamp.unit));
             }
             case UUID -> uuidType();
+            case FLOAT16 -> float16Type();
```

Comment:
Should this be test covered?

cc @Fokko

Reply:
It's always a good idea to add some tests. I think some examples from Parquet-MR can be ported:

- https://github.com/apache/parquet-mr/pull/1142/files#diff-0f04d9a45bc14ae3d19a06c14cc84171b2454dfd1edad62be30a23fce53f2191
- https://github.com/apache/parquet-mr/pull/1235/

Reply:
@Fokko , from Trino user perspective, what would happen if we didn't have the line `case FLOAT16 -> float16Type();`?

Reply:
Maybe let's throw instead ;)

Reply:
Then the compiler would complain since not all cases are covered. You could also decide to not support Float16 for Trino.

Reply:
> Maybe let's throw instead ;)

That also works :)

Reply:
@Fokko I believe that @findepi asked about having `case FLOAT16 -> throw new UnsupportedOperationException()` instead

Reply:
Sure, that works until someone tries to read/write Float16 fields 👍

Reply:
That's exactly the kind of info i am looking for -- 
from Trino user perspective, what does it mean to "read/write Float16 fields"?

Can we have a test with reading parquet file with float16 field?
That would exercise this code path.
And also ensure we're not missing anything else (like floatToIntBits or something)

Reply:
We should throw exception here. From writer perspective this is a performance optimization, but we cannot adopt it in near future as new things in parquet standard are almost never widely supported. We would just be risking creating data which many applications cannot read for a limited perf gain.
From a reader perspective, it would be nice to support reading it. But it would require more than just the one line change here, so it should be handled in a separate issue. I don't think we should bother implementing it unless we start to run into users actually needing it. We went years without supporting another FLOAT/DOUBLE related encoding called BYTE_STREAM_SPLIT and no one cared. We support it now, but I'm yet to run into anyone actually using it.

Reply:
https://github.com/trinodb/trino/pull/21916